meetings/2024-01-24.md

```diff
@@ -0,0 +1,116 @@
+# Node.js Technical Steering Committee (TSC) Meeting 2024-01-24
+
+## Links
+
+* **Recording**:  <https://www.youtube.com/watch?v=KzBhROQFcDE>
+  * seems like the recording dropped at 34mins, not sure why, unfortunately because
+    there was a lot of good discussion
+* **GitHub Issue**: <https://github.com/nodejs/TSC/issues/1495>
+
+## Present
+
+* Antoine du Hamel @aduh95 (voting member)
+* Geoffrey Booth @GeoffreyBooth (voting member)
+* James Snell @jasnell (voting member)
+* Joyee Cheung @joyeecheung (voting member)
+* Chengzhong Wu @legendecas (voting member)
+* Michael Dawson @mhdawson (voting member)
+* Moshe Atlow @MoLow (voting member)
+* Richard Lau @richardlau (voting member)
+* Ruy Adorno @ruyadorno (voting member)
+* Myles Borins @MylesBorins (regular member)
+
+## Agenda
+
+### Announcements
+
+* Antoine - 2 successful collaborator nominations - lemire and zcbenz
+
+### CPC and Board Meeting Updates
+
+*Extracted from **tsc-agenda** labeled issues and pull requests from the **nodejs org** prior to the meeting.
+
+### nodejs/node
+
+* enable corepack by default [#50963](https://github.com/nodejs/node/issues/50963)
+  * Myles, speaking as fellow collaborator
+    * Believe questions are being framed incorrectly
+    * Question is about enabling yarn, pump and having them just work
+    * When looking at what Node.js is shipping only 10% is coming from supported Node.js
+      versions, lots still come from older versions
+    * yarn and pnpm have large enough downloads,
+    * npm don’t want to support/integrate with corepack as it does not fit with view of how
+      package managers should be integrated
+    * corepack was started due to context that limited willingness to add yarn/pnpm to Node.js,
+      but extended beyond that.
+    * believe we should make yarn and pnpm just work, but that does not necessarily mean
+      corepack
+    * should define governance for supporting clients
+  * Michael
+    * Need to factor in work on the project to support any particular solution
+  * Richard, if npm was not currently bundled we probably would not bundle, so if we define
```

Comment:
@richardlau @MylesBorins @mhdawson I was unable to attend the meeting but the part I don't understand is with regards to @mcollina 's point from last week.

Currently there is exactly one package manager hosting a public registry (with a few mirrors), all other package managers are npm clients that talk to the npm registry. npm can (but doesn't) break other clients but that's out off kindness.

The situation isn't equitable because exactly one client is footing the server bill, not because one client is bundled - and I suspect that had we made the choice today npm would still be bundled or we would ship our own registry (which would incur high costs and would require the project to foot the bill instead of fGitHub).

Reply:
There's obviously more than one public registry, npm is "just" the most popular one especially for the JS ecosystem. Let's not start discussing npm intents, whether it's kindness or whatever, we don't know that (unless you do?), and I don't think it's relevant. FWIW I agree with Richard, if npm wasn't bundled today, we would probably not bundle it; my guess is that npm themselves would distribute `node` – and that would probably work better for everyone.

(You probably already know that, this PR is about keeping notes of what folks say, so this discussion should not block the PR from landing)

Reply:
My point was more that if we were to define criteria today for bundling a package manager with Node.js it is likely that npm wouldn't meet that criteria (see for example, historic tension around npm's lack of long-term support) (FTR I have no idea if any of the other package managers would meet such criteria either). Personally I do not consider that a reason by itself to suggest not bundling npm now as it would be too disruptive to the ecosystem.

Reply:
Agreed that npm has a release management that is at odds with the one with follow, and it would likely not meet that bar. Specifically around the fact that they use September as the month to ship their major, because we go LTS in October.

@benjamingr can you refer to what I said last week that was not clear?

Reply:
My 2 cents of clarifying what @mcollina said. It is that because the npm client comes from and is supported by GitHub who hosts the registry, it will always have the distinction of being the "official" client in terms of the provider of the registry (GitHub). Nothing that the node.js project does will change the benefits that distinction brings to the npm client.  For example if something in the registry does not work with the npm client, then GitHub should work to fix that either in the registry or the client. If something does not work with some of the other package managers then they might consider it a problem in the client and not be interested in making changes in the registry even if that means it will be a lot of work for the client to fix the issue.

Reply:
Thanks @mhdawson, exactly that.